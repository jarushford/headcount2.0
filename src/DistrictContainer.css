.district-container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
  grid-gap: 15px;
  margin: 40px;
}