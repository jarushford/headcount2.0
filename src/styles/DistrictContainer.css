.district-container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
  grid-gap: 25px;
  margin: 40px 40px 0 40px;
  padding-bottom: 50px;
}