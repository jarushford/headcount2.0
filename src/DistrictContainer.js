import React from 'react'
import District from './District'
import './styles/DistrictContainer.css'
import PropTypes from 'prop-types'
import  { uid } from 'react-uid'

export default function DistrictContainer({ districts, showAll, compareDistrict }) {
  if (!showAll) { districts = districts.slice().splice(0, 12) }
  const districtArray = districts.map(district => {
    return <District districtData={district} key={uid(district)}
      compareDistrict={compareDistrict} />
  })

  return (
    <div className='district-container'>
      { districtArray }
    </div>
  )
}

DistrictContainer.propTypes = {
  districts: PropTypes.arrayOf(PropTypes.object)
}