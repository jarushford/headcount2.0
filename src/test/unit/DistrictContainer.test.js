import React from 'react';
import DistrictContainer from '../../DistrictContainer';
import { shallow } from 'enzyme'


describe('App', () => {
  const data = [ { location: 'Colorado', stats: { 2006: 0 } } ]
  it('should match the snapshot with all data passed in for showLess', () => {
    const wrapper = shallow(<DistrictContainer districts={data}
      showAll={false}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with all data passed in for showAll', () => {
    const wrapper = shallow(<DistrictContainer districts={data}
      showAll={true}/>)
    expect(wrapper).toMatchSnapshot()
  })

})