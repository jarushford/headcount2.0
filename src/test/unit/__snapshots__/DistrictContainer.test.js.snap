// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictContainer
    districts={
        Array [
            Object {
              "location": "Colorado",
              "stats": Object {
                "2006": 0,
              },
            },
          ]
    }
    showAll={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <District
          districtData={
                    Object {
                              "location": "Colorado",
                              "stats": Object {
                                "2006": 0,
                              },
                            }
          }
/>,
      ],
      "className": "district-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "uid1",
        "nodeType": "function",
        "props": Object {
          "districtData": Object {
            "location": "Colorado",
            "stats": Object {
              "2006": 0,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <District
            districtData={
                        Object {
                                    "location": "Colorado",
                                    "stats": Object {
                                      "2006": 0,
                                    },
                                  }
            }
/>,
        ],
        "className": "district-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "function",
          "props": Object {
            "districtData": Object {
              "location": "Colorado",
              "stats": Object {
                "2006": 0,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match the snapshot with all data passed in for showAll 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictContainer
    districts={
        Array [
            Object {
              "location": "Colorado",
              "stats": Object {
                "2006": 0,
              },
            },
          ]
    }
    showAll={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <District
          districtData={
                    Object {
                              "location": "Colorado",
                              "stats": Object {
                                "2006": 0,
                              },
                            }
          }
/>,
      ],
      "className": "district-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "uid1",
        "nodeType": "function",
        "props": Object {
          "districtData": Object {
            "location": "Colorado",
            "stats": Object {
              "2006": 0,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <District
            districtData={
                        Object {
                                    "location": "Colorado",
                                    "stats": Object {
                                      "2006": 0,
                                    },
                                  }
            }
/>,
        ],
        "className": "district-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "function",
          "props": Object {
            "districtData": Object {
              "location": "Colorado",
              "stats": Object {
                "2006": 0,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match the snapshot with all data passed in for showLess 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictContainer
    districts={
        Array [
            Object {
              "location": "Colorado",
              "stats": Object {
                "2006": 0,
              },
            },
          ]
    }
    showAll={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <District
          districtData={
                    Object {
                              "location": "Colorado",
                              "stats": Object {
                                "2006": 0,
                              },
                            }
          }
/>,
      ],
      "className": "district-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "uid1",
        "nodeType": "function",
        "props": Object {
          "districtData": Object {
            "location": "Colorado",
            "stats": Object {
              "2006": 0,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <District
            districtData={
                        Object {
                                    "location": "Colorado",
                                    "stats": Object {
                                      "2006": 0,
                                    },
                                  }
            }
/>,
        ],
        "className": "district-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "function",
          "props": Object {
            "districtData": Object {
              "location": "Colorado",
              "stats": Object {
                "2006": 0,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
